pytorch_pfn_extras/nn/modules/ensure_shape.py

```diff
@@ -0,0 +1,115 @@
+from typing import Optional, Tuple
+
+import torch
+
+
+class Ensure(torch.nn.Module):
+    """Module to check the shape of a tensor.
+
+    Args:
+       shape: Tuple with the desired shape. If the input tensor shape
+           is not compatible, `ValueError` will be raised. If `None` is set
+           as a dimension value, that dimension will be ignored.
+       dtype: Checks if the `dtype` of the input thensor matches the
+           provided one.
+       broadcastable: Check if the shapes are compatible using broadcasting
+           rules.
+       can_cast: Check if the input tensor can be casted to the provided type.
+    """
+
+    def __init__(
+            self,
+            *,
+            shape: Optional[Tuple[int]] = None,
```

Comment:
mypy failed in CI as `shape` may contain None.

```suggestion
            shape: Optional[Tuple[Optional[int]]] = None,
```